apps/meteor/server/lib/findUsersOfRoomOrderedByRole.ts

```diff
@@ -0,0 +1,188 @@
+import type { IUser, IRole } from '@rocket.chat/core-typings';
+import { Subscriptions } from '@rocket.chat/models';
+import { escapeRegExp } from '@rocket.chat/string-helpers';
+import type { Document, FilterOperators } from 'mongodb';
+
+import { settings } from '../../app/settings/server';
+
+type FindUsersParam = {
+	rid: string;
+	status?: FilterOperators<string>;
+	skip?: number;
+	limit?: number;
+	filter?: string;
+	sort?: Record<string, any>;
+	rolesInOrder?: IRole['_id'][];
+	exceptions?: string[];
+	extraQuery?: Document[];
+};
+
+type UserWithRoleData = IUser & {
+	roles: IRole['_id'][];
+};
+
+export async function findUsersOfRoomOrderedByRole({
```

Comment:
Can you add some explains with some decent amount of data to check on the perf of the aggregation? :please: